x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/ExactMatchProcessor.java

```diff
@@ -56,49 +56,41 @@
         final String value = ingestDocument.getFieldValue(enrichKey, String.class, ignoreMissing);
         if (value == null) {
             return ingestDocument;
         }
 
-        // TODO: re-use the engine searcher between enriching documents from the same write request
-        Tuple<IndexMetaData, Engine.Searcher> tuple = searchProvider.apply(EnrichPolicy.getBaseName(policyName));
-        String enrichKeyField = getEnrichKeyField(tuple.v1());
-
-        try (Engine.Searcher engineSearcher = tuple.v2()) {
-            if (engineSearcher.getDirectoryReader().leaves().size() == 0) {
-                return ingestDocument;
-            } else if (engineSearcher.getDirectoryReader().leaves().size() != 1) {
-                throw new IllegalStateException("enrich index must have exactly a single segment");
-            }
-
-            final LeafReader leafReader = engineSearcher.getDirectoryReader().leaves().get(0).reader();
-            final Terms terms = leafReader.terms(enrichKeyField);
-            if (terms == null) {
-                throw new IllegalStateException("enrich key field does not exist");
-            }
-
-            final TermsEnum tenum = terms.iterator();
-            if (tenum.seekExact(new BytesRef(value))) {
-                PostingsEnum penum = tenum.postings(null, PostingsEnum.NONE);
-                final int docId = penum.nextDoc();
-                assert docId != PostingsEnum.NO_MORE_DOCS : "no matching doc id for [" + enrichKey + "]";
-                assert penum.nextDoc() == PostingsEnum.NO_MORE_DOCS : "more than one doc id matching for [" + enrichKey + "]";
-
-                // TODO: The use of _source is temporarily until enrich source field mapper has been added (see PR #41521)
-                Document document = leafReader.document(docId, Set.of(SourceFieldMapper.NAME));
-                BytesRef source = document.getBinaryValue(SourceFieldMapper.NAME);
-                assert source != null;
-
-                final BytesReference encoded = new BytesArray(source);
-                final Map<String, Object> decoded =
-                    XContentHelper.convertToMap(encoded, false, XContentType.SMILE).v2();
-                for (EnrichSpecification specification : specifications) {
-                    Object enrichValue = decoded.get(specification.sourceField);
-                    // TODO: add support over overwrite option (like in SetProcessor)
-                    ingestDocument.setFieldValue(specification.targetField, enrichValue);
-                }
-            }
+        TermQueryBuilder termQuery = new TermQueryBuilder(enrichKey, value);
+        // TODO: Use a custom transport action instead of the search API
+        SearchSourceBuilder searchBuilder = new SearchSourceBuilder();
+        searchBuilder.fetchSource(specifications.stream().map(s -> s.sourceField).toArray(String[]::new), null);
+        searchBuilder.query(termQuery);
```

Comment:
In order to optimize this as much as possible:
1) wrap `termQuery` in a `ConstantScoreQueryBuilder` and set `searchBuilder.trackScores` to `false`, so that scoring is disabled.
2) set `searchBuilder .size` to `1`